AutomationPracticeMainPage.java

```diff
@@ -0,0 +1,45 @@
+package pageobject.AutomationPracticeSite;
+
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import pageobject.AbstractPage;
+
+public class AutomationPracticeMainPage extends AbstractPage {
+
+    @FindBy(xpath = "(//div[@class='product-container'])[7]")
+    private WebElement productItem;
+
+    @FindBy(xpath = "(//a[@class='product-name'][@title='Printed Chiffon Dress'])[1]")
+    private WebElement productTitle;
+
+    @FindBy(xpath = "//a[@title='Printed Chiffon Dress']")
+    private WebElement productImageItem;
+
+    @FindBy(xpath = "(//span[@class='price product-price'])[14]")
+    private WebElement productPrice;
+
+    public AutomationPracticeMainPage(WebDriver webDriver) {
+        super(webDriver);
+    }
+
+    public boolean isPresentOfProductItem() {
+        webDriverWait.until(ExpectedConditions.visibilityOf(productItem));
```

Comment:
what will happen if `until ` is not completed successfully?

Reply:
If the until method does not work, then an error will be thrown to the console.

Reply:
well, 'error to console' is tricky description. the question - are you sure code that follows that until() method will be executed?

Reply:
I just debugged this method and you were right.
boolean item = productItem.isDisplayed(); - this line of code doesn't work.
Therefore, there is no need to use until() method here.